lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -6,213 +6,97 @@ defmodule RealtimeWeb.RealtimeChannel do
 
   require Logger
 
+  alias Realtime.Helpers
+
   alias DBConnection.Backoff
+
   alias Phoenix.Tracker.Shard
-  alias RealtimeWeb.{ChannelsAuthorization, Endpoint, Presence}
-  alias Realtime.{GenCounter, RateCounter, PostgresCdc, SignalHandler, Tenants}
-
-  import Realtime.Helpers, only: [cancel_timer: 1, decrypt!: 2]
-
-  defmodule Assigns do
-    @moduledoc false
-    defstruct [
-      :tenant,
-      :log_level,
-      :rate_counter,
-      :limits,
-      :tenant_topic,
-      :pg_sub_ref,
-      :pg_change_params,
-      :postgres_extension,
-      :claims,
-      :jwt_secret,
-      :tenant_token,
-      :access_token,
-      :postgres_cdc_module,
-      :channel_name
-    ]
 
-    @type t :: %__MODULE__{
-            tenant: String.t(),
-            log_level: atom(),
-            rate_counter: RateCounter.t(),
-            limits: %{
-              max_events_per_second: integer(),
-              max_concurrent_users: integer(),
-              max_bytes_per_second: integer(),
-              max_channels_per_client: integer(),
-              max_joins_per_second: integer()
-            },
-            tenant_topic: String.t(),
-            pg_sub_ref: reference() | nil,
-            pg_change_params: map(),
-            postgres_extension: map(),
-            claims: map(),
-            jwt_secret: String.t(),
-            tenant_token: String.t(),
-            access_token: String.t(),
-            channel_name: String.t()
-          }
-  end
+  alias Realtime.GenCounter
+  alias Realtime.PostgresCdc
+  alias Realtime.RateCounter
+  alias Realtime.SignalHandler
+  alias Realtime.Tenants
+
+  alias RealtimeWeb.ChannelsAuthorization
+  alias RealtimeWeb.Endpoint
+  alias RealtimeWeb.Presence
```

Comment:
Two lines of aliases have grown to eight. 

Hey, @supabase/dashbit, what's your recommendation on using [multi-aliases](https://elixir-lang.org/getting-started/alias-require-and-import.html#multi-aliasimportrequireuse)? Would you advise against aliasing multiple modules at once?

Reply:
We're definitely using multi-aliases on our projects. Nothing wrong with that. :)

The only concern is right now if you grep by `Realtime.GenCounter` you'll find this alias declaration but when it is changed to `alias Realtime.{GenCounter, ...}`, especially when it's spread across multiple lines, you no longer will be. FWIW, tools like ElixirLS, `mix xref`, "compilation tracers", etc are still able to see the "expanded" aliases.

I don't think it is a concern for most projects actually, I don't think it ever came up in projects I was directly involved in. There was an interesting twitter discussion about exactly this yesterday so you may want to check it out: https://twitter.com/ZacharyNDean/status/1705632083715019167.

As an aside and this is my personal opinion I often feel like we're using aliases too much. (My colleagues would sigh because I often would be the one that _did_ add a bunch of aliases.) It's fine to refer to modules by their full names. One easy thing people forget about is on apps (as opposed to libraries) the top-level name doesn't matter that much. If it's annoying (which I'd slightly agree with) to type `Realtime.Repo` and `RealtimeWeb.Endpoint`, I'd have called these `RT.Repo` and `RTWeb.Endpoint`. :)

Reply:
@wojtekmach thank you for the explanation! We had a similar discussion to the one in the tweet above, so I decided to clarify this thing 🙏

Reply:
lol this discussion will probably never end. I'm in the one-line-per-alias camp but at this point I don't particularly care. Should just setup a credo rule or something per project and just do what it says. @abc3 you can be the decider for Realtime :D

Reply:
No strong opinions either. With other projects with thousands of files it became annoying to just ctrl+f an alias but we're not close to that 😆

> As an aside and this is my personal opinion I often feel like we're using aliases too much.

This!!! A lot of times when we're aliasing a module with two words it feels weird and not really necessary 

As for a Credo rule, there's one for the "option1" in that tweet as an optional https://hexdocs.pm/credo/Credo.Check.Readability.MultiAlias.html